klippy/extras/palette2.py

```diff
@@ -0,0 +1,610 @@
+# Palette 2 MMU support, Firmware 9.0.9 and newer supported only!
+#
+# Copyright (C) 2021 Clifford Roche <clifford.roche@gmail.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+import os
+import threading
+import time
```

Comment:
FYI, it's not a good idea to use `time.time()` in klippy - it's preferable to use `reactor.monotonic()` instead.

Reply:
Noted, I'll make the change.